app/lib/common/utils/utils.dart

```diff
@@ -184,6 +187,53 @@ String randomString() {
   return base64UrlEncode(values);
 }
 
+Future<void> openAvatar(
+  BuildContext context,
+  WidgetRef ref,
+  String roomId,
+) async {
+  final membership = await ref.watch(roomMembershipProvider(roomId).future);
+  final canUpdateAvatar = membership?.canString('CanUpdateAvatar') == true;
+  final avatarInfo = ref.watch(roomAvatarInfoProvider(roomId));
```

Comment:
always using `ref.read` in function calls, as we are not actually setting up any watcher and data isn't being executed if things changed. Using `read` clarifies that this is the case.

```suggestion
  final membership = await ref.read(roomMembershipProvider(roomId).future);
  final canUpdateAvatar = membership?.canString('CanUpdateAvatar') == true;
  final avatarInfo = ref.read(roomAvatarInfoProvider(roomId));
```